lib/internal/async_hooks.js

```diff
@@ -116,7 +116,7 @@ function callbackTrampoline(asyncId, cb, ...args) {
     emitBeforeNative(asyncId);
 
   let result;
-  if (typeof domain_cb === 'function') {
+  if (asyncId === 0 && typeof domain_cb === 'function') {
```

Comment:
small nit: I find it a bit confusing to use `asyncId === 0` here but no at the other places in this function.

Reply:
@flarna I’ve made the other places also use an explicit comparison to `0`, is that what you had in mind?

Reply:
Yes, either this or remove it here.